core/templates/dev/head/pages/exploration_player/HelpCardService.js

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to track the information about Help Card
+ */
+
+oppia.factory('HelpCardService', [
+  function() {
+    var localHelpCard = {
+      helpCardHtml: null,
+      hasContinueButton: false
+    }
+
+    return {
+      getHtmlHelpCard: function() {
```

Comment:
I think just `getHelpCard` and `setHelpCard` should so. I'm not sure why you have html in it. I'm not opposed to keeping it, but if you do then change `clearHelpCard` to `clearHtmlHelpCard`.

Reply:
I'm also not in favour of having Html in the name but the scope object of SupplementCardDirective has the property called helpCardHtml, so to have a commonality I used that name. But I'll change it.

Reply:
Create a domain object for HelpCardContent, as @tjiang11 suggested above. Then call this getHelpCardContent() and setHelpCardContent().

That said, I don't really understand why we have this service in the first place. If you had a directive for the help card then I could see the help card populating its values from the service ... but then you'll need some sort of event to trigger the update if you don't want it to continuously poll the service...

So, let's take a step back. Could you explain what the point of this PR is, and how it solves the original problem?